File: core/src/main/java/in/koreatech/koin/core/util/ToastUtil.java
package in.koreatech.koin.core.util;

import android.content.Context;
import android.widget.Toast;

import androidx.annotation.StringRes;

/**
 * Created by hyerim on 2018. 4. 9....
 */
public class ToastUtil {
    public static void makeShortToast(Context context, String message) {
        Toast.makeText(context, message, Toast.LENGTH_SHORT).show();
    }

    public static void makeLongToast(Context context, String message) {
        Toast.makeText(context, message, Toast.LENGTH_LONG).show();
    }

    public static void makeShortToast(Context context, @StringRes int message) {
        Toast.makeText(context, message, Toast.LENGTH_SHORT).show();
    }

    public static void makeLongToast(Context context, @StringRes int message) {
        Toast.makeText(context, message, Toast.LENGTH_LONG).show();
    }
}

File: core/util/src/main/java/in/koreatech/core/util/ToastUtil.kt
package `in`.koreatech.core.util

import android.content.Context
import android.widget.Toast
import androidx.annotation.StringRes
import androidx.fragment.app.Fragment

fun Context.makeShortToast(message: String) {
    Toast.makeText(this, message, Toast.LENGTH_SHORT).show()
}

fun Context.makeLongToast(message: String) {
    Toast.makeText(this, message, Toast.LENGTH_LONG).show()
}

fun Context.makeShortToast(@StringRes resId: Int) {
    Toast.makeText(this, resId, Toast.LENGTH_SHORT).show()
}

fun Context.makeLongToast(@StringRes resId: Int) {
    Toast.makeText(this, resId, Toast.LENGTH_LONG).show()
}

fun Fragment.makeShortToast(message: String) {
    requireContext().makeShortToast(message)
}

fun Fragment.makeLongToast(message: String) {
    requireContext().makeLongToast(message)
}

fun Fragment.makeShortToast(@StringRes resId: Int) {
    requireContext().makeShortToast(resId)
}

fun Fragment.makeLongToast(@StringRes resId: Int) {
    requireContext().makeLongToast(resId)
}